### PR TITLE
Paginate moderations page without losing filtering

### LINKED ISSUE #2442: 🐛 [Bug] reports query is too expensive and times out often
> [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-12-06T09:22:22Z" title="Friday, December 6th 2019, 10:22:22 am +01:00">Dec 6, 2019</time>_
_Closed <time datetime="2019-12-09T20:23:13Z" title="Monday, December 9th 2019, 9:23:13 pm +01:00">Dec 9, 2019</time>_
---

## :bug: Bugreport

The reports query fetches all reports, which have nested `FiledReports`, nested `reviews`,  nested `resources`, which have nested attributes as well. 
This makes it an extremely expensive request. Leading to time outs.

### Steps to reproduce the behavior
1. log in as admin
2. go to `/moderation`
3. see if the reports load
4. ...
5. Profit

### Expected behavior

We'd expect that `reports` would be shown

### Additional context

@roschaefer suggested we could do `eager` loading of certain resources.
We could also try pagination

### PR DESCRIPTION
**Pull request by Tirokk**
> [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-12-09T15:38:29Z" title="Monday, December 9th 2019, 4:38:29 pm +01:00">Dec 9, 2019</time>_
_Merged <time datetime="2019-12-09T20:23:13Z" title="Monday, December 9th 2019, 9:23:13 pm +01:00">Dec 9, 2019</time>_
---

Co-authored-by: Tirokk <wolle.huss@pjannto.com>

### Issues

- fixes #2442

### CODE REVIEW

**Comment by Tirokk** [comment]
> [cypress[bot]](https://github.com/apps/cypress) **Authored by [cypress[bot]](https://github.com/apps/cypress)**
_<time datetime="2019-12-09T16:09:15Z" title="Monday, December 9th 2019, 5:09:15 pm +01:00">Dec 9, 2019</time>_

---



<p align="center">
  <br />
  
</p>

---

### Test summary
<a href="https://dashboard.cypress.io/projects/qa7fe2/runs/2763/specs"></a> <b title="Passed">48</b>  &bull; <a href="https://dashboard.cypress.io/projects/qa7fe2/runs/2763/failures"></a> <b title="Failed">0</b>  &bull;  <b title="Pending">0</b> &bull;  <b title="Skipped">0</b>

---

### Run details
| | |
| --- | --- |
| **Project** | Human-Connection |
| **Status** | Passed |
| **Commit** | 5271337b49 |
| **Started** | Dec 9, 2019 7:30 PM |
| **Ended** | Dec 9, 2019 7:41 PM |
| **Duration** | 11:23 <a href="https://on.cypress.io/parallelization" title="Enable parallelization to reduce run duration">💡</a> |
| **OS** | Linux Ubuntu Linux - 16.04 |
| **Browser** | Chromium 78  |

[View run in Cypress Dashboard ➡️](https://dashboard.cypress.io/projects/qa7fe2/runs/2763/specs)



---

This comment has been generated by **cypress-bot** as a result of this project's GitHub integration settings. You can manage this integration in [this project's settings in the Cypress Dashboard](https://dashboard.cypress.io/projects/qa7fe2/settings)

File: webapp/graphql/Moderation.js [hunk 1]
```diff
@@ -82,9 +82,6 @@ export const reportsListQuery = () => {
             name
             disabled
             deleted
-            followedByCount
-            contributionsCount
-            commentedCount
```
Comment (Tirokk) [inline webapp/graphql/Moderation.js:87]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-12-09T17:07:42Z" title="Monday, December 9th 2019, 6:07:42 pm +01:00">Dec 9, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -82,9 +82,6 @@ export const reportsListQuery = () => {
             name
             disabled
             deleted
-            followedByCount
-            contributionsCount
-            commentedCount
```


Awww yeah. That's what I wanted to do in my next PR. I'm going to rebase my code on that.

File: webapp/components/User/User.vue [hunk 2]
```diff
@@ -26,7 +26,7 @@
         </div>
       </nuxt-link>
     </template>
-    <template slot="popover">
+    <template slot="popover" v-if="showCounts">
```
Comment (Tirokk) [inline webapp/components/User/User.vue:29]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-12-09T17:09:45Z" title="Monday, December 9th 2019, 6:09:45 pm +01:00">Dec 9, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -26,7 +26,7 @@
         </div>
       </nuxt-link>
     </template>
-    <template slot="popover">
+    <template slot="popover" v-if="showCounts">
```


I would say `showPopover` no?

Comment (Tirokk) [reply webapp/components/User/User.vue:29]: > [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-12-09T17:49:58Z" title="Monday, December 9th 2019, 6:49:58 pm +01:00">Dec 9, 2019</time>_

---


yes, that's better

**Review by Mogge — APPROVED** [review]
Looks really good! I will rebase my own code on top of that.

git

**Pull request closed without merging**